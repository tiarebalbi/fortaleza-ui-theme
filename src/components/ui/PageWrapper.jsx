// @flow

import * as React from 'react';
import styled from 'styled-components';
import type { PageWrapperType } from '../../types';

const Style = styled.div`
  padding-bottom: 20px;

  h1 {
    color: #4a4a4a;
    font-size: 20px;
    margin-bottom: 0.5em;
  }

  p {
    font-size: 12px;
    font-weight: normal;
  }

  .wrapper-content {
    background: rgba(255, 255, 255, 0.89);
    border-radius: 6px;
    box-shadow: 0 1px 10px 0 rgba(39, 52, 125, 0.1);
    padding: 12px;
    height: 56px;

    a,
    button {
      float: left;
      margin-right: 10px;
    }

    a.right,
    button.right {
      float: right;
      margin-left: 10px;
    }
  }
`;

const PageWrapper = (props: PageWrapperType) => (
  <Style>
    <h1>{props.title}</h1>
    {props.description && <p>{props.description}</p>}
    {props.children && <div className="wrapper-content">{props.children}</div>}
  </Style>
);

export default PageWrapper;
