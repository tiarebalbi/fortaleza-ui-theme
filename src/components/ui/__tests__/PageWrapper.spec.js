import React from 'react';
import PageWrapper from '../PageWrapper';
import { mount } from 'enzyme';

describe('ui/PageWrapper', () => {
  const wrapper = mount(<PageWrapper title="Page">a</PageWrapper>);

  it('should be stateless', () => {
    expect(wrapper.instance()).toBeNull();
  });

  it('should render as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should hold description if present', () => {
    wrapper.setProps({ description: 'Sample Description' });
    expect(wrapper.find('p')).toHaveText('Sample Description');
  });

  it('should not have wrapper-content if children is undefined', () => {
    const local = mount(<PageWrapper title="Page" />);
    expect(local.find('div.wrapper-content')).not.toExist();
  });
});
