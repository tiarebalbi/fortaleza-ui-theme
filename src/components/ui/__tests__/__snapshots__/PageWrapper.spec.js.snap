// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/PageWrapper should render as expected 1`] = `
<PageWrapper
  title="Page"
>
  <styled.div>
    <div
      className="sc-bdVaJa fgBQjw"
    >
      <h1>
        Page
      </h1>
      <div
        className="wrapper-content"
      >
        a
      </div>
    </div>
  </styled.div>
</PageWrapper>
`;
